### PR TITLE
[editor] Combine the local and api strategies into a new SqlAnalyzer

### PR DESCRIPTION
**Pull request by JohanAhlen**
With this change it will now always call the api to retrieve top joins.

### CODE REVIEW

File: desktop/core/src/desktop/js/catalog/optimizer/optimizer.ts [hunk 1]
```diff
@@ -90,13 +89,13 @@ export const API_STRATEGY = 'api';
 
 const createOptimizer = (connector: Connector): Optimizer => {
   // TODO: Remove window.OPTIMIZER_MODE and hardcoded { optimizer: 'api' } when 'connector.optimizer_mode' works.
-  if ((<hueWindow>window).OPTIMIZER_MODE === LOCAL_STRATEGY) {
-    return new LocalStrategy(connector);
+  if (
```
Comment (romainr) [inline desktop/core/src/desktop/js/catalog/optimizer/optimizer.ts:92]: :+1:

File: desktop/core/src/desktop/js/catalog/optimizer/SqlAnalyzer.ts [hunk 2]
```diff
@@ -80,79 +84,56 @@ export default class LocalStrategy implements Optimizer {
...
-              joinCols: [{ columns: [path + '.' + key.name, key.to] }],
-              tables: [path].concat(key.to.split('.', 2).join('.')),
-              joinType: 'join'
-            }))
-          });
-        })
-        .catch(reject);
-    });
+  fetchTopJoins(options: PopularityOptions): CancellablePromise<TopJoins> {
+    return this.apiStrategy.fetchTopJoins(options);
```
Comment (romainr) [inline desktop/core/src/desktop/js/catalog/optimizer/SqlAnalyzer.ts:88]: Is there a way to keep the Js Join logic all the time (code that was removed) in this strategy too?
(and it gets augmented by apiStrategy.fetchTopJoins)

i.e. optimize.getJoin is missing a lot of JOINs/dialects, js Join is based on FKs of left autocomplete so get a lot of solid ones for free.

Comment (romainr) [reply desktop/core/src/desktop/js/catalog/optimizer/SqlAnalyzer.ts:88]: (same idea for the Risk too)

Comment (JohanAhlen) [reply desktop/core/src/desktop/js/catalog/optimizer/SqlAnalyzer.ts:88]: Ah, indeed, my bad. Will join.

Comment (JohanAhlen) [reply desktop/core/src/desktop/js/catalog/optimizer/SqlAnalyzer.ts]: Updated with merging of local top joins and risks.

**Review by romainr — APPROVED** [review]
Uber bueno!

(and no worries about dup suggestions for now)

**Pull request merged**